### PR TITLE
making flask more user-friendly

### PR DESCRIPTION
**Pull request by kzidane**

### CODE REVIEW

File: files/etc/profile.d/ide50.sh [hunk 1]
```diff
@@ -94,17 +94,17 @@ flask()
...
         while test ${#} -gt 0
         do
-            if [[ "$1" =~ ^--host= ]]; then
+            if echo "$1" | grep -q "^--host="; then
                 host="$1"
-            elif [[ "$1" =~ ^--port= ]]; then
+            elif echo "$1" | grep -q "^--port="; then
                 port="$1"
-            elif [[ "$1" =~ ^--with(out)?-threads$ ]]; then
+            elif echo "$1" | grep -q "^--with\(out\)\?-threads$"; then
```
Comment (dmalan) [inline files/etc/profile.d/ide50.sh]: Are the `\` necessary if not using `-E` with `grep`? Cf. https://www.gnu.org/software/grep/manual/html_node/Basic-vs-Extended.html.

Comment (kzidane) [inline files/etc/profile.d/ide50.sh]: Yup, but I'll change to `egrep` to simplify.

> In basic regular expressions the meta-characters **‘?’**, ‘+’, ‘{’, ‘|’, **‘(’, and ‘)’ lose their special meaning**; instead **use the backslashed versions ‘\\?’**, ‘\\+’, ‘\\{’, ‘\\|’, **‘\\(’, and ‘\\)’**.

File: files/etc/profile.d/ide50.sh [hunk 2]
```diff
@@ -67,9 +67,49 @@ export FLASK_APP=application.py
...
+                    options="$1"
+                else
+                    options="$options $1"
+                fi
+            fi
+            shift
+        done
+
+        # spawn flask
+        script --flush --quiet --return /dev/null --command "FLASK_APP=$FLASK_APP FLASK_DEBUG=$FLASK_DEBUG flask run $host $port $threads $options" |
```
Comment (dmalan) [inline files/etc/profile.d/ide50.sh]: We should quote each of these vars, just in case one ends up having spaces.

Comment (kzidane) [inline files/etc/profile.d/ide50.sh]: Done. Values for other env vars throughout are quoted per #76!

File: files/etc/profile.d/ide50.sh [hunk 3]
```diff
@@ -67,9 +67,49 @@ export FLASK_APP=application.py
...
+                    options="$1"
+                else
+                    options="$options $1"
+                fi
+            fi
+            shift
+        done
+
+        # spawn flask
+        script --flush --quiet --return /dev/null --command 'FLASK_APP="$FLASK_APP" FLASK_DEBUG="$FLASK_DEBUG" flask run $host $port $threads $options' |
```
Comment (dmalan) [inline files/etc/profile.d/ide50.sh]: I forget, will $vars definitely get interpolated by `sh` when inside single quotes? Or do we need to use `"` and `\"`?

Comment (kzidane) [inline files/etc/profile.d/ide50.sh]: It should be interpolated in the `flask` command itself not in the `script` command, so the literal command that should end up being run by `script` is:

```
FLASK_APP="$FLASK_APP" FLASK_DEBUG="$FLASK_DEBUG" flask run $host $port $threads $options
```

Comment (kzidane) [inline files/etc/profile.d/ide50.sh]: Can use `\"`, if you'd like? Just thought it's simpler this way.

**Comment by kzidane** [comment]
Included per #104!

**Pull request closed without merging**